### PR TITLE
Build an rlib for shadow-rs

### PR DESCRIPTION
**Pull request by stevenengler**
This adds doc tests in shadow-rs.

### CODE REVIEW

File: src/main/Cargo.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ publish = false
 
 [lib]
 path = "lib.rs"
-crate-type = ["staticlib"]
+crate-type = ["rlib", "staticlib"]
```
Comment (sporksmith) [inline src/main/Cargo.toml:9]: Probably worth a comment here that the `rlib` is there to enable doc-tests

File: src/main/core/support/units.rs [hunk 2]
```diff
@@ -2,13 +2,17 @@
 Types for parsing/deserializing unit values.
 
 ```
+# use shadow_rs::core::support::units::*;
+# use std::str::FromStr;
+# fn main() {
```
Comment (sporksmith) [inline src/main/core/support/units.rs:7]: I don't think you need the `fn main()` ?

**Review by sporksmith — APPROVED** [review]

File: src/main/core/support/units.rs (hunk 2 above)
Comment (stevenengler) [reply src/main/core/support/units.rs]: Oops, fixed!

File: src/main/Cargo.toml (hunk 1 above)
Comment (stevenengler) [reply src/main/Cargo.toml:10]: Added a comment.

**Pull request merged**